pkg/dynamic/client.go

```diff
@@ -141,14 +152,56 @@ func (c *knDynamicClient) ListSources(types ...WithType) (*unstructured.Unstruct
 
 		if len(sList.Items) > 0 {
 			// keep a track if we found source objects of different types
-			numberOfsourceTypesFound++
+			numberOfSourceTypesFound++
+			sourceList.Items = append(sourceList.Items, sList.Items...)
+			sourceList.SetGroupVersionKind(sList.GetObjectKind().GroupVersionKind())
+		}
+	}
+	// Clear the Group and Version for list if there are multiple types of source objects found
+	// Keep the source's GVK if there is only one type of source objects found or requested via --type filter
+	if numberOfSourceTypesFound > 1 {
+		sourceList.SetGroupVersionKind(schema.GroupVersionKind{Group: "", Version: "", Kind: "List"})
+	}
+	return &sourceList, nil
+}
+
+// ListSourcesUsingGVKs returns list of available source objects using given list of GVKs
+func (c *knDynamicClient) ListSourcesUsingGVKs(gvks *[]schema.GroupVersionKind, types ...WithType) (*unstructured.UnstructuredList, error) {
```

Comment:
I wonder what is the best function behavior when `gvks` is `nil`: a) to return a `nil` list or b) to ignore `gvks` and use `types` to filter. The current behavior is a).
Yet when `types` is `nil`, the current behavior is b) to ignore `types` and use `gvks` to filter, not a) to return a `nil` list. It makes me a little weird. 
Of course it doesn't affect the current functions. For future usage, maybe change to the same behavior when either of these two parameters is `nil`.

Reply:
We use `types` only to subset the found source types. If gvks list is nil, it returns nil. Does this answer your question ?